ArduCopter/GCS_Mavlink.cpp

```diff
@@ -37,25 +37,11 @@ MAV_MODE GCS_MAVLINK_Copter::base_mode() const
     // only get useful information from the custom_mode, which maps to
     // the APM flight mode and has a well defined meaning in the
     // ArduPlane documentation
-    switch (copter.flightmode->mode_number()) {
-    case Mode::Number::AUTO:
-    case Mode::Number::AUTO_RTL:
-    case Mode::Number::RTL:
-    case Mode::Number::LOITER:
-    case Mode::Number::AVOID_ADSB:
-    case Mode::Number::FOLLOW:
-    case Mode::Number::GUIDED:
-    case Mode::Number::CIRCLE:
-    case Mode::Number::POSHOLD:
-    case Mode::Number::BRAKE:
-    case Mode::Number::SMART_RTL:
+    if ((copter.pos_control != nullptr) && copter.pos_control->is_active_xy()) {
         _base_mode |= MAV_MODE_FLAG_GUIDED_ENABLED;
```

Comment:
This makes our answer here marginally worse.

My reading of that attribute is we should only set it actually chasing a waypoints "flies waypoints / mission items".

Not a blocker, but we really should cut this down to AUTO, AUTO_RTL, RTL and SMART_RTL at some stage.